renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -525,6 +580,109 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
     return replacements;
   }
 
+  uint32_t GetEventFlags(const VulkanCreationInfo::Pipeline &p, const VulkanRenderState &pipestate)
+  {
+    uint32_t flags = 0;
+
+    // Culling
+    {
+      if(p.cullMode != VK_CULL_MODE_NONE)
+        flags |= TestEnabled_Culling;
+
+      if(p.cullMode == VK_CULL_MODE_FRONT_AND_BACK)
+        flags |= TestMustFail_Culling;
+    }
+
+    // Depth and Stencil tests.
+    {
+      if(p.depthBoundsEnable)
+        flags |= TestEnabled_DepthBounds;
+
+      if(p.depthTestEnable)
+      {
+        if(p.depthCompareOp != VK_COMPARE_OP_ALWAYS)
+          flags |= TestEnabled_DepthTesting;
+        if(p.depthCompareOp == VK_COMPARE_OP_NEVER)
+          flags |= TestMustFail_DepthTesting;
+      }
+
+      if(p.stencilTestEnable)
+      {
+        if(p.front.compareOp != VK_COMPARE_OP_ALWAYS || p.back.compareOp != VK_COMPARE_OP_ALWAYS)
+          flags |= TestEnabled_StencilTesting;
+
+        if(p.front.compareOp == VK_COMPARE_OP_NEVER && p.back.compareOp == VK_COMPARE_OP_NEVER)
+          flags |= TestMustFail_StencilTesting;
+        else if(p.front.compareOp == VK_COMPARE_OP_NEVER && p.cullMode == VK_CULL_MODE_BACK_BIT)
+          flags |= TestMustFail_StencilTesting;
+        else if(p.cullMode == VK_CULL_MODE_FRONT_BIT && p.back.compareOp == VK_COMPARE_OP_NEVER)
+          flags |= TestMustFail_StencilTesting;
+      }
+    }
+
+    // Scissor
+    {
+      bool inRegion = false;
+      bool inAllRegions = true;
+      // Do we even need to know viewerport here?
+      const VkRect2D *pScissors;
+      uint32_t scissorCount;
+      if(p.dynamicStates[VkDynamicViewport])
```

Comment:
Is this looking at the wrong state? I'd have expected `VkDynamicScissor`.

Reply:
You are right, thanks!